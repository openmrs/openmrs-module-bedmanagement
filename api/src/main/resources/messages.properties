${project.parent.artifactId}.title=bedmanagement Module
${project.parent.artifactId}.admissionLocations=Admission Locations
${project.parent.artifactId}.bedTypes=Bed Types
${project.parent.artifactId}.bedTags=Bed Tags