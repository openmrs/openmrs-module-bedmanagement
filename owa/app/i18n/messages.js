export default {
    en: {
        ADMISSION_LOCATIONS: 'Admission Locations',
        BED_TYPES: 'Bed Types',
        BED_TAGS: 'Bed Tags',
        LOGOUT: 'Logout',
        CHOOSE_LANGUAGE: 'Choose language',
        ROW_REQUIRED_MSG: 'Row is required field',
        COLUMN_REQUIRED_MSG: 'Column is required field',
        ROW_SHOULD_GREATER_THAN_ZERO: 'Row value should be greater than 0',
        ROW_SHOULD_GREATER_THAN_LAYOUT_ROW: 'Row value should not be greater than layout row size',
        COLUMN_SHOULD_GREATER_THAN_ZERO: 'Column value should be greater than 0 and less than 11',
        COLUMN_SHOULD_GREATER_THAN_LAYOUT_COLUMN: 'Row value should not be greater than layout row size',
        FIX_ERROR_MSG: 'Fix error before submit',
        ADMISSION_LOCATION_SAVE_MSG: 'Admission location save successfully',
        BED_LAYOUT_SAVE_MSG: 'Admission location bed layout save successfully',
        BED_TAG_SAVE_MSG: 'Bed Tag save successfully',
        BED_TYPE_SAVE_MSG: 'Bed Type save successfully',
        BED_SAVE_MSG: 'Bed save successfully',
        DELETE_ADMISSION_LOCATION_CONFIRM_MESSAGE:
            'Are you sure you want to delete admission location {location_name}?',
        DELETE_BED_CONFIRM_MESSAGE: 'Are you sure you want to delete bed number {bed_number}?',
        DELETE_BED_TAG_CONFIRM_MSG: 'Are you sure you want to delete bed Tag {bed_tag_name}?',
        DELETE_BED_TYPE_CONFIRM_MSG: 'Are you sure you want to delete bed type {bed_type_name}?',
        DELETE_SUCCESSFULLY: 'Delete successfully',
        SUCCESS: 'Success',
        ERROR: 'Error',
        INFO: 'Info',
        SAVE: 'Save',
        SAVING: 'Saving...',
        CANCEL: 'Cancel',
        ADD: 'Add',
        EDIT: 'Edit',
        DELETE: 'Delete',
        ADD_BED: 'Add Bed',
        ROWS: 'Rows',
        ROW: 'Row',
        COLUMNS: 'Columns',
        COLUMN: 'Column',
        SET_LAYOUT: 'Set Layout',
        LOCATION: 'Location',
        BED: 'Bed',
        BED_NUMBER: 'Bed Number',
        BED_TYPE: 'Bed Type',
        WARD: 'Ward',
        PARENT_LOCATION: 'Parent Location',
        NONE: 'None',
        NAME: 'Name',
        DESCRIPTION: 'Description',
        DISPLAY_NAME: 'Display Name',
        ACTION: 'Action',
        ADD_NEW: 'Add New',
        BED_TAG: 'Bed Tag',
        EXISTING_BED_TYPES: 'Existing Bed Types',
        EXISTING_BED_TAGS: 'Existing Bed Tags'
    }
};
