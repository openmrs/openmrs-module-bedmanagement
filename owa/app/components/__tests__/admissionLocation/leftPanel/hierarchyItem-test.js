import React from 'react';
import {shallowToJson} from 'enzyme-to-json';
import {shallow, mount} from 'enzyme';
import {IntlProvider} from 'react-intl';

import HierarchyItem from 'components/admissionLocation/leftPanel/hierarchyItem';
import hierarchyFunctionMock from 'components/__mocks__/hierarchyFunctions-mock';
import messages from 'i18n/messages';

const intlProvider = new IntlProvider({locale: 'en', messages: messages['en']}, {});
const {intl} = intlProvider.getChildContext();
const testData = {
    props: {
        hierarchyFunction: hierarchyFunctionMock,
        admissionLocations: hierarchyFunctionMock.getAdmissionLocations()
    },
    context: {
        intl: intl
    }
};

describe('HierarchyItem', () => {
    it('Renders correctly General ward hierarchy', () => {
        const hierarchyItem = shallow(
            <HierarchyItem
                key="baf7bd38-d225-11e4-9c67-080027b662ec"
                isParentOpen={true}
                hierarchyFunction={testData.props.hierarchyFunction}
                admissionLocation={testData.props.admissionLocations['baf7bd38-d225-11e4-9c67-080027b662ec']}
            />,
            {context: testData.context}
        );

        expect(shallowToJson(hierarchyItem)).toMatchSnapshot();
    });

    it('Renders correctly Labour ward hierarchy', () => {
        const hierarchyItem = shallow(
            <HierarchyItem
                isParentOpen={true}
                hierarchyFunction={testData.props.hierarchyFunction}
                admissionLocation={testData.props.admissionLocations['bb0e512e-d225-11e4-9c67-080027b662ec']}
            />,
            {context: testData.context}
        );

        expect(
            hierarchyItem
                .find('ul')
                .text()
                .trim()
        ).toBe('Labour Ward');
        expect(shallowToJson(hierarchyItem)).toMatchSnapshot();
    });

    it('Should trigger event handler', () => {
        const spyOnClickTitle = jest.spyOn(HierarchyItem.prototype, 'onClickTitle');
        const spyOnClickIcon = jest.spyOn(HierarchyItem.prototype, 'onClickIcon');
        const spySetState = jest.spyOn(testData.props.hierarchyFunction, 'setState');
        const spySetAdmissionLocationIsOpen = jest.spyOn(
            testData.props.hierarchyFunction,
            'setAdmissionLocationIsOpen'
        );
        const spyToggleIsOpen = jest.spyOn(testData.props.hierarchyFunction, 'toggleIsOpen');

        const hierarchyItem = mount(
            <HierarchyItem
                isParentOpen={true}
                hierarchyFunction={testData.props.hierarchyFunction}
                admissionLocation={testData.props.admissionLocations['bb0e512e-d225-11e4-9c67-080027b662ec']}
            />,
            {context: testData.context}
        );

        hierarchyItem.find('span').simulate('click');
        expect(spyOnClickTitle).toHaveBeenCalled();
        expect(spySetAdmissionLocationIsOpen).toHaveBeenCalledWith('bb0e512e-d225-11e4-9c67-080027b662ec', true);
        expect(spySetState).toHaveBeenCalledWith({
            activeUuid: 'bb0e512e-d225-11e4-9c67-080027b662ec',
            activePage: 'listing',
            pageData: {}
        });

        hierarchyItem.find('i').simulate('click');
        expect(spyOnClickIcon).toHaveBeenCalled();
        expect(spyToggleIsOpen).toHaveBeenCalledWith('bb0e512e-d225-11e4-9c67-080027b662ec', false);
    });
});
