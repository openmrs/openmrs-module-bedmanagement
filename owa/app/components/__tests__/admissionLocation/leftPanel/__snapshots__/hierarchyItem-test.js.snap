// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HierarchyItem Renders correctly General ward hierarchy 1`] = `
<li
  className=""
>
  <ul>
    <li>
      <i
        aria-hidden="true"
        className="fa fa-plus-square"
        onClick={[Function]}
      />
      <span
        className="active"
        onClick={[Function]}
      >
         
        General Ward
         
      </span>
    </li>
    <HierarchyItem
      admissionLocation={
        Object {
          "description": "General Ward room number 2",
          "isHigherLevel": false,
          "isOpen": false,
          "name": "General Ward - Room 2",
          "parentAdmissionLocationUuid": "baf7bd38-d225-11e4-9c67-080027b662ec",
          "uuid": "e48fb2b3-d490-11e5-b193-0800270d80ce",
        }
      }
      hierarchyFunction={
        Object {
          "getActiveUuid": [Function],
          "getAdmissionLocations": [Function],
          "setActiveUuid": [Function],
          "setAdmissionLocationIsOpen": [Function],
          "setState": [Function],
          "toggleIsOpen": [Function],
        }
      }
      isParentOpen={false}
      key="e48fb2b3-d490-11e5-b193-0800270d80ce"
    />
    <HierarchyItem
      admissionLocation={
        Object {
          "description": "General Ward of second floor",
          "isHigherLevel": false,
          "isOpen": false,
          "name": "General Ward - floor-2",
          "parentAdmissionLocationUuid": "baf7bd38-d225-11e4-9c67-080027b662ec",
          "uuid": "baf83667-d225-11e4-9c67-080027b662ec",
        }
      }
      hierarchyFunction={
        Object {
          "getActiveUuid": [Function],
          "getAdmissionLocations": [Function],
          "setActiveUuid": [Function],
          "setAdmissionLocationIsOpen": [Function],
          "setState": [Function],
          "toggleIsOpen": [Function],
        }
      }
      isParentOpen={false}
      key="baf83667-d225-11e4-9c67-080027b662ec"
    />
  </ul>
</li>
`;

exports[`HierarchyItem Renders correctly Labour ward hierarchy 1`] = `
<li
  className=""
>
  <ul>
    <li>
      <i
        aria-hidden="true"
        className="fa fa-caret-right"
        onClick={[Function]}
      />
      <span
        className=""
        onClick={[Function]}
      >
         
        Labour Ward
         
      </span>
    </li>
  </ul>
</li>
`;
